lib/zabbix.js

```diff
@@ -26,30 +26,68 @@ var debug;
 var flushInterval;
 var zabbixHost;
 var zabbixPort;
-var zabbixSender;
+var zabbixCmd;
+var zabbixSender = null;
+var senderModule;
 
 var zabbixStats = {};
 
 var post_stats = function zabbix_post_stats(statString) {
   if (zabbixHost) {
-    if (debug) {
-      util.log(statString);
-    }
+    if(zabbixCmd) {
+      // we're spawning a child process to run the shell zabbix_sender command
+      try {
+        var zabbixExec = proc.exec(zabbixCmd, function (error, stdout, stderr) {
+          if (error && error.code !== 0 && debug) {
+            util.log(zabbixSender + ': ' + stderr);
+          }
+        });
 
-    try {
-      var zabbixExec = proc.exec(zabbixCmd, function(error, stdout, stderr) {
-        if (error && error.code !== 0 && debug) {
-          util.log(zabbixSender + ': ' + stderr);
+        zabbixExec.stdin.write(statString);
+        zabbixExec.stdin.end();
+      } catch (e) {
+        if (debug) {
+          util.log(e);
         }
-      });
+        zabbixStats.last_exception = Math.round(new Date().getTime() / 1000);
+      }
+    }
+    else {
+      // we're using the module, not a shell command, so no child process
+      try {
+        var Sender = new senderModule({host: zabbixHost, with_timestamps: true, items_host: '68.226.99.8'});
```

Comment:
This has an IP hardcoded? Generally it'd be better to rework this section into a function call that accepts the same arguments for both exec and node, decided based on config in the init below.

Reply:
Whoops!  Agreed. That setting hardcoded was totally unintended. Not sure how I left that in there.